packages/shorebird_cli/lib/src/commands/release_new/aar_releaser.dart

```diff
@@ -0,0 +1,190 @@
+  String get buildNumber => argResults['build-number'] as String;
+
+  Set<Arch> get architectures => (argResults['target-platform'] as List<String>)
+      .map(
+        (platform) => AndroidArch.availableAndroidArchs
+            .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+      )
+      .toSet();
+
+  @override
+  ReleaseType get releaseType => ReleaseType.aar;
+
+  @override
+  bool get requiresReleaseVersionArg => true;
+
+  @override
+  Future<void> assertPreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+      );
+    } on PreconditionFailedException catch (e) {
+      exit(e.exitCode.code);
+    }
+  }
+
+  @override
+  Future<void> assertArgsAreValid() async {
+    if (!argResults.wasParsed('release-version')) {
+      logger.err('Missing required argument: --release-version');
+      exit(ExitCode.usage.code);
+    }
+  }
+
+  @override
+  Future<FileSystemEntity> buildReleaseArtifacts() async {
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+    final buildAppBundleProgress =
+        logger.progress('Building aar with Flutter $flutterVersionString');
+
+    try {
+      await artifactBuilder.buildAar(
+        buildNumber: buildNumber,
+        targetPlatforms: architectures,
+      );
+    } catch (e) {
+      logger.err('Failed to build aar: $e');
+      exit(ExitCode.software.code);
+    }
+
+    buildAppBundleProgress.complete(
+      'Built aar with Flutter $flutterVersionString',
+    );
+
+    // Copy release AAR to a new directory to avoid overwriting with
+    // subsequent patch builds.
+    final sourceLibraryDirectory = Directory(
+      ShorebirdAndroidArtifacts.aarLibraryPath,
+    );
+    final targetLibraryDirectory = Directory(
+      p.join(shorebirdEnv.getShorebirdProjectRoot()!.path, 'release'),
+    );
+    await copyPath(
+      sourceLibraryDirectory.path,
+      targetLibraryDirectory.path,
+    );
+
+    return targetLibraryDirectory;
+  }
+
+  @override
+  Future<String> getReleaseVersion({
+    required FileSystemEntity releaseArtifactRoot,
+  }) async {
+    return argResults['release-version'] as String;
+  }
+
+  @override
+  Future<void> uploadReleaseArtifacts({
+    required Release release,
+    required String appId,
+  }) async {
+    final extractAarProgress = logger.progress('Creating artifacts');
+    final extractedAarDir = await shorebirdAndroidArtifacts.extractAar(
+      packageName: shorebirdEnv.androidPackageName!,
+      buildNumber: buildNumber,
+      unzipFn: extractFileToDisk,
+    );
+    extractAarProgress.complete();
+
+    await codePushClientWrapper.createAndroidArchiveReleaseArtifacts(
+      appId: appId,
+      releaseId: release.id,
+      platform: releaseType.releasePlatform,
+      aarPath: ShorebirdAndroidArtifacts.aarArtifactPath(
+        packageName: shorebirdEnv.androidPackageName!,
```

Comment:
Is it safe to force unwrap here?

Reply:
I think so, but added a step to validate preconditions to check. Good catch!